### PR TITLE
Xsoar linter validations

### PR DESCRIPTION
**Pull request by orhovy**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
https://github.com/demisto/etc/issues/27484

## Description
added additional validations :
* main function as an warning
* sys.exit - moved to certified partner from base
* demisto.log as an warning.
* no sleep statements for not long running integrations.
* added support for checking if an integration is long running.

## Screenshots
Paste here any images that will help the reviewer

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: demisto_sdk/commands/lint/resources/pylint_plugins/base_checker.py [hunk 1]
```diff
@@ -44,12 +43,13 @@ def _print_checker(self, node):
         except Exception:
             pass
 
-    def _sys_exit_checker(self, node):
-        try:
-            if node.func.attrname == 'exit' and node.func.expr.name == 'sys' and node.args and node.args[0].value != 0:
-                self.add_message("sys-exit-exists", node=node)
-        except Exception:
-            pass
+    def _sleep_checker(self, node):
```
Comment (bakatzir) [inline demisto_sdk/commands/lint/resources/pylint_plugins/base_checker.py:46]: if someone `from time import sleep` he can only have `sleep(x)` in the code, let's make sure we cover this as well.

File: demisto_sdk/commands/lint/tests/test_pylint_plugin/base_checker_test.py [hunk 2]
```diff
@@ -103,79 +103,47 @@ def test_function():
             self.checker.visit_call(node_a)
 
 
-class TestSysExitChecker(pylint.testutils.CheckerTestCase):
+class TestSleepChecker(pylint.testutils.CheckerTestCase):
```
Comment (bakatzir) [inline demisto_sdk/commands/lint/tests/test_pylint_plugin/base_checker_test.py:106]: let's add a case of a sleep for a long running integration.

**Review by bakatzir — COMMENTED** [review]
Very good.
Let's set a short time with @ShahafBenYakir regarding the warnings.
Let's see how they look in the circle logs, and maybe take on swiftly a task to comment it in the PR.
@avidan-H will probably be able to assist us with that.

File: demisto_sdk/commands/lint/tests/test_pylint_plugin/base_checker_test.py (hunk 2 above)
Comment (orhovy) [reply demisto_sdk/commands/lint/tests/test_pylint_plugin/base_checker_test.py:106]: there is a case for those. its in the xsoar_linter_integration_tests.
test_plugin folder contain only tests on detecting the required string in the code correctly using pylint.
in the integration tests - I am testing different support levels, long running integrations and it tests end to end the xsoar linter activity on valid and invalid files.

File: demisto_sdk/commands/lint/resources/pylint_plugins/base_checker.py (hunk 1 above)
Comment (orhovy) [reply demisto_sdk/commands/lint/resources/pylint_plugins/base_checker.py:46]: you are right , added a check for that and a test.

**Review by bakatzir — COMMENTED** [review]
I'm feeling confident with the additions.
Let's make sure we have a workable action item on how to comment the warning in the PR before merging this.
even better would be to test that PR with this to see that it is working smoothly together.

File: demisto_sdk/commands/lint/resources/pylint_plugins/base_checker.py [hunk 3]
```diff
@@ -1,35 +1,34 @@
...
 
+# -------------------------------------------- Messages for all linters ------------------------------------------------
+
+
 class CustomBaseChecker(BaseChecker):
     __implements__ = IAstroidChecker
     name = "base-checker"
     priority = -1
-    msgs = Msg_XSOAR_linter.get('base_checker')
+    msgs = base_msg
```
Comment (bakatzir) [inline demisto_sdk/commands/lint/resources/pylint_plugins/base_checker.py:24]: can you explain why this is needed?

File: demisto_sdk/commands/lint/tests/test_pylint_plugin/certified_partner_level_checker_test.py [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+            - String of a code part which is being examined by pylint plugin.
+        When:
+            - sys.exit exists in the code.
+        Then:
+            - Ensure that the correct message id is being added to the message errors of pylint
+        """
+        _, node_b, _ = astroid.extract_node("""
+            def test_function(): #@
+                sys.exit(1) #@
+                return True #@
```
Comment (bakatzir) [inline demisto_sdk/commands/lint/tests/test_pylint_plugin/certified_partner_level_checker_test.py:28]: what is the usage of `#@`?

**Review by bakatzir — APPROVED** [review]
I'm good with the changes. please run it on content to make sure we don't have any `sleep` statements in non long running integrations.

File: demisto_sdk/commands/lint/resources/pylint_plugins/base_checker.py (hunk 3 above)
Comment (orhovy) [reply demisto_sdk/commands/lint/resources/pylint_plugins/base_checker.py:24]: This is the list of messages which are relevent to this checker. every time an error is found this error is added to pylint's list of errors. as each checker inherents from Pylints BaseChecker, it has to have it.

File: demisto_sdk/commands/lint/tests/test_pylint_plugin/certified_partner_level_checker_test.py (hunk 4 above)
Comment (orhovy) [reply demisto_sdk/commands/lint/tests/test_pylint_plugin/certified_partner_level_checker_test.py:28]: #@ is used from pylint testutils(which those tests inherent from) and means that this row will be evaluated( which means we will have the type of node of this row as a return value from astroid.extract_node

**Comment by orhovy** [comment]
Added error messages for exit() and quit()

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,6 +2,17 @@
 * Added support for layoutscontainer in **init** contribution flow.
 * Added a validation for tlp_color param in feeds in **validate** command.
 * Fixed an issue where **update-release-notes** was failing with a wrong error message when no pack or input was given.
+* Added new code validations to `XSOAR-linter`.
+    * As error messages:
+        * sleep statments for non longrunning integrations.
```
Comment (bakatzir) [inline CHANGELOG.md:7]:
```suggestion
        * Longer than10 seconds sleep statements for non long running integrations.
```

**Review by bakatzir — APPROVED** [review]

**Pull request merged**